packages/web-react/src/components/Modal/useModalDialogStyleProps.ts

```diff
@@ -0,0 +1,45 @@
+import { ElementType, CSSProperties } from 'react';
+import { ModalDialogCSSHeight, ModalDialogCSSHeightBreakpoints, ModalDialogProps } from '../../types';
+
+interface CustomizedHeightCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+const setCustomHeight = (
+  baseVarName: string,
+  propValue: ModalDialogCSSHeight | ModalDialogCSSHeightBreakpoints,
+): CustomizedHeightCSSProperties => {
+  const customizedHeightStyle: CustomizedHeightCSSProperties = {};
+  if (typeof propValue === 'object' && propValue !== null) {
+    Object.keys(propValue).forEach((key) => {
+      const suffix = key === 'mobile' ? '' : `-${key}`;
+      const propName = `--${baseVarName}${suffix}`;
+
+      customizedHeightStyle[propName] = propValue[key as keyof ModalDialogCSSHeightBreakpoints]?.toString();
+    });
+  } else {
+    const propName = `--${baseVarName}`;
+    customizedHeightStyle[propName as keyof CustomizedHeightCSSProperties] = propValue?.toString();
+  }
+
+  return customizedHeightStyle;
+};
+
+export function useModalDialogStyleProps<E extends ElementType>(props: ModalDialogProps<E>) {
```

Comment:
I think we are using mostly the arrow functions.

Reply:
fulltext search:
`export const use` - 44 times
`export function use` - 28 times

So yeah, "mostly" :D